storybookjs/storybook
# Add surplus support

Closes #8485 

## What I did

Added [Surplus](https://github.com/adamhaile/surplus) support.

![image](https://user-images.githubusercontent.com/885648/67891004-0cac0180-fb52-11e9-93ba-d5dd2487afb6.png)


## How to test

Well, storybook is pretty hairy to hack on, so this is how I've been testing:

1. Build everything using `yarn bootstrap`
2. Run verdacio via `yarn bootstrap` -> `Run local registry`
3. Publish all modules
4. `mkdir /tmp/test-sb-surplus && cd /tmp/test-sb-surplus`
5. `npm install @storybook/cli && node_modules/.bin/sb init`
6. Answer `y` to the initialization prompt; select `SURPLUS` from the list
7. After it initializes the package.json, there appears to be a bug somewhere that dumps `{"type": "buffer", "data": [...]}` which, for some reason, is the output of `npm init -y` (not the package.json file, but the literal console output) - not sure what is up with that. I simply `rm package.json` and move on with my day.
8. Run `node_modules/.bin/start-storybook -p 6006`

- Is this testable with Jest or Chromatic screenshots?

I'm sure. I need some guidance here.

- Does this need a new example in the kitchen sink apps?

I'm not a huge user of Storybook so I'm not a good person to answer this. I _think_ it needs a whole new kitchen sink - not just a new example.

- Does this need an update to the documentation?

Yes, I have not updated any docs (yet) in this PR. I'd like to make sure all of the functional aspects are there and available.
